### PR TITLE
More control of the amount of items displayed for node, user, and switches.

### LINKED ISSUE #381: Search pagination control (how many results)
Please add an option to select how many results are displayed.

### LINKED ISSUE #699: Need a pagination framework
Call it perl on rails.
Will make julian happy.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

More control of the amount of items displayed for node, user, and switches.
## Impacts

Admin GUI
## Dependencies

List the PRs or other factors on which this PR depends
## Issue

Fixes #699
Fixes #381
## NEWS file entries

Enhancements
++++++++++++
- Added the option of how many results are displayed on the Node page
- Added the option of how many results are displayed on the User page
- Added the option of how many results are displayed on the Switch
## Delete branch after merge

NO

### CODE REVIEW

**Comment by fdurand** [comment]
Tested and works, but there is just a detail when i want to have per example 100 devices on the default node page (without any filter) it doesn't work.
Another point, maybe more than 100 for the node page should be usefull like 500, 1000.

**Comment by rouzier** [comment]
That is fixed and I expanded the available options

**Comment by fdurand** [comment]
Just tested and works as expected.

**Comment by fdurand** [comment]
But if i select 1000 and reg all the devices (action->Unreg), it return 25 devices and i am not able to click anymore on action.

**Comment by julsemaan** [comment]
`Added the option of how many results are displayed on the Page page`

I always wondered where that `Page page` was hidden

File: html/pfappserver/lib/pfappserver/Base/Controller/Crud/Pagination.pm [hunk 1]
```diff
@@ -18,6 +18,8 @@ use HTTP::Status qw(:constants is_error is_success);
 use MooseX::MethodAttributes::Role;
 use namespace::autoclean;
 use HTML::FormHandler::Params;
+use POSIX qw(ceil);
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Base/Controller/Crud/Pagination.pm]: Look like useless

File: html/pfappserver/lib/pfappserver/Model/Config/Switch.pm [hunk 2]
```diff
@@ -15,9 +15,11 @@ pfappserver::Model::Config::Switch;
 use Moose;
 use namespace::autoclean;
 use pf::config::cached;
+use pf::util qw(calc_page_count);
 use pf::config;
 use pf::ConfigStore::Switch;
 use HTTP::Status qw(:constants is_error is_success);
+use POSIX qw(ceil);
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Model/Config/Switch.pm]: Look like useless

File: html/pfappserver/lib/pfappserver/Model/Config/Switch.pm [hunk 3]
```diff
@@ -87,30 +91,38 @@ Search the config from query
...
+            if($start <= $found_count && $found_count <= $end) {
+                push @items,$item;
+            }
+            $found_count++;
+        }
     }
+    my $pageCount = calc_page_count($found_count, $perPage);
+#    my $count = @items;
+#    $end = $count - 1 if $end >= $count;
+#    @items = @items[$start..$end];
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Model/Config/Switch.pm]: Remove

**Pull request merged**